server/src/api/customers.ts

```diff
@@ -43,19 +41,35 @@ customerRouter.get(
 );
 
 /**
- * Create a customer if they do not already exist.
+ * Endpoint for the "logging in" of customers.
+ * If the customer exists, return the customer details in the
+ * response body with code 200.
+ * Else, add the customer and return the added customer details
+ * in the response body with code 201.
  */
 customerRouter.post(
   '/',
   async (req: Request, res: Response, next: NextFunction) => {
     const customerData: CustomerPayload = req.body;
 
     try {
-      const customer = await customerService.addCustomer(customerData);
-      const resourceUrl = `${process.env.SERVER_URL}/customers/${customer.id}`;
+      // Customers are unique by their gpayId, so we will retrieve
+      // the customer using their gpay Id.
+      const existingCustomer = await customerService.getCustomerWithGpayId(
+        customerData.gpayId
+      );
+      if (existingCustomer !== null) {
+        const resourceUrl = `${process.env.SERVER_URL}/customers/${existingCustomer.id}`;
+        res.setHeader('Content-Location', resourceUrl);
```

Comment:
Nice! You checked that this case doesn't need the 'Location' header but still requires the 'Content-Location'.